### DISCUSSION TITLE
Common General Feature to provide WebAPI Query &  Dataverse Query for every tool in XRM Toolbox

### DISCUSSION THREAD

**Discussion by nickgsoft** [MscrmTools/XrmToolBox#1133] [Ideas]
@MscrmTools 

- Technically, most XRM Toolbox tools are making webapi queries to CRM instances.. or SDK Link Queries in general.
- If that statement is true, then can we have some inbuilt feature to expose and convert Tool specific WebAPI calls and add feature to show console window with exact api queries made by every tool while execution? it may allow transparency to developer? or allow couple button on every tool to "show sample WebAPI Query used by tool" and "Show sample Dataverse SQL queries made by tool" ?

Thoughts?  If this is doable in most tool, this tool will become similar to XRMPostman tool or XRMDataverseTool? may be.. or is this over thinking?

If needed we can set up online session and deep dive the details and share screens...

**Comment by nickgsoft** [comment]
@MscrmTools  thoughts on this topic? i am willing to discuss over a teams call if you like?

**Comment by MscrmTools** [comment]
I have started to review how this could be achieved but I did not find anything so far that would help to implement this feature
